src/Pulsar.Client/Internal/ConsumerImpl.fs

```diff
@@ -355,9 +381,13 @@ type ConsumerImpl internal (consumerConfig: ConsumerConfiguration, clientConfig:
                                 Log.Logger.LogDebug("{0} Ignoring message from before the startMessageId: {1}", prefix, startMessageId)
                             else
                                 let message = { rawMessage with MessageId = msgId } |> decompress
+
                                 if not hasWaitingChannel then
                                     incomingMessages.Enqueue(message)
+                                    storeDeadLetter message
```

Comment:
storeDeadLetter in both branches, can it be moved to one level hight?

Reply:
Done.